edit/edit.go

```diff
@@ -248,15 +248,40 @@ func InsertAtEnd(stmt []build.Expr, expr build.Expr) []build.Expr {
 
 // FindRuleByName returns the rule in the file that has the given name.
 // If the name is "__pkg__", it returns the global package declaration.
-func FindRuleByName(f *build.File, name string) *build.Rule {
+func FindRuleByName(f *build.File, pkg string, name string) *build.Rule {
 	if name == "__pkg__" {
 		return PackageDeclaration(f)
 	}
 	i := IndexOfRuleByName(f, name)
 	if i != -1 {
 		return &build.Rule{Call: f.Stmt[i].(*build.CallExpr)}
 	}
+	return UseImplicitName(f, pkg, name)
+}
+
+// UseImplicitName returns the rule in the file if it meets these conditions:
+// - It is the only rule in the file.
+// - The passed package and rule names match.
+func UseImplicitName(f *build.File, pkg string, rule string) *build.Rule {
```

Comment:
You should be able to use filepath.Base(filepath.Dir(f.Path)) to get the last component of the package name, rather than needing to pass it in explicitly.

This also means you don't need to do the string slicing below :)

(Should probably also error if f.Path == "BUILD", because I don't believe an empty implicit name is allowed)